### ISSUE TITLE
CBuilder XE8 install issue

### ISSUE THREAD

**Issue by mikecollins999** [TurboPack/LockBox3#2]
Hay Guys (Roman).

Great work on porting lockbox to XE8 - I've been desperately awaiting its return.

I'm struggling to install into C++Builder XE8.  I have unpacked the components into $(BCB)\Components\LockBox3.  The install notes mention adding in the path to the hpp folder into the C++ System Include paths which I have done ($(BCB)\Components\LockBox3\hpp).  it wasn't clear if, for C++Builder you also had to add in the \lockBox3\run path also (I haven't).

I open up the design and run-time CBuilder packages and try to install the design-time package - it fails reporting:

[ilink32 Error] Fatal: Unable to open file 'LOCKBOX3CR.BPI'

I try to compile also and I get the same message.  I have searched the lockbox3 folders and there is no sign of the .bpi file although it is listed in the Requires branch in the IDE.

Also, out of interest, do I need to add the path to the \hpp to all of the profiles (Win32, 64, iOS, Android etc in order to use lockbox on all these platforms.

Many thanks

Mike

**Comment by romankassebaum** [comment]
After you compile the runtime package you should be able to find the file LockBox3CR.bpi in the folder

C:\Users\Public\Documents\Embarcadero\Studio\16.0\DCP\

Can you find it there?

**Comment by mikecollins999** [comment]
Hi Roman,

Sorry for the confusion, I didn't realise that you had to compile run-time
libraries first.

I have managed to compile and install the libraries but when I add the
components to a new CBuilder XE8 VCL application and compile I get the
following compile error:

[bcc32 Error] uTPLb_CryptographicLibrary.hpp(174): E2370 Simple type name
expected

relating to line 174 in uTPLb_CryptographicLibrary.hpp:

 __property TOnGenerateKeyFunc OnCustomCipherGenerateKey =
{read=FOnGenerateKeyFunc, write=FOnGenerateKeyFunc};

I have attached the compile output.

There are a couple of other things that I am confused about (sorry if my
questions are basic but I'm sure others might be struggling with the same
thing and if I can clarify them, maybe I can help others. I would also like
to try and help to drive the documentation of LockBox forward as they are
brilliant but often overlooked components):

1) When setting the system include paths (for BCB), is the path to the HPP
folder the ONLY one that needs to be set?

2) Does this need to be set for all platforms i.e. iOS, Android, Win32,
Win64?

3) When installing the run-time and design-time packages, do the run-time
packages need to be built / compiled for each platform that you want to
deploy them onto or is it sufficient to only build them for Win32 (I know
this might seem obvious but it isn't clear to me).

4) In following with this, is it necessary to compile the run-time and
design-time packages in debug & release mode?

5) Finally - is there any documentation, help files, videos etc to get
started?  LockBox 3 is very different to LockBox 2 and I'm struggling a bit
to get my head around how to use the components.  The About mentions
http://lockbox.seanbdurkin.id.au/LB3Help_Home but the link doesn't work.

I'm by no means an expert in anything but I would be more than happy to try
and add to this project in terms of documentation or re-wording the readme
to make it clearer to us layman :-)

Many thanks in advance and best regards

Mike Collins



On Tue, May 12, 2015 at 11:32 AM, Roman Kassebaum <notifications@github.com>
wrote:

> After you compile the runtime package you should be able to find the file
> LockBox3CR.bpi in the folder
>
> C:\Users\Public\Documents\Embarcadero\Studio\16.0\DCP\
>
> Can you find it there?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TurboPack/LockBox3/issues/2#issuecomment-101232262>.
>

﻿Checking project dependencies...
Compiling Project1.cbproj (Debug, Win32)
bcc32 command line for "Unit1.cpp"
  c:\program files\embarcadero\studio\16.0\bin\bcc32.exe -D_DEBUG -DUSEPACKAGES -n.\Win32\Debug -I"c:\program 
  files\embarcadero\studio\16.0\include\windows\vcl";"c:\program files\embarcadero\studio\16.0\include";"c:\program 
  files\embarcadero\studio\16.0\include\dinkumware";"c:\program files\embarcadero\studio\16.0\include\windows\crtl";"c:\program 
  files\embarcadero\studio\16.0\include\windows\sdk";"c:\program files\embarcadero\studio\16.0\include\windows\rtl";"c:\program 
  files\embarcadero\studio\16.0\include\windows\vcl";"c:\program files\embarcadero\studio\16.0\include\windows\fmx";"C:\Program 
  Files\FastReports\LibD22";"C:\Program Files\Raize\CS5\Lib\RS-XE8\Win32";"c:\program files\embarcadero\studio\16.0\Components\LockBox3\hpp";
  C:\Users\Public\Documents\Embarcadero\Studio\16.0\hpp\Win32 -y -Q -k -r- -c -tR -tM -tU -tW -C8 -o.\Win32\Debug\Unit1.obj -w-par -Od -v -vi- 
  -H=.\Win32\Debug\Project1.pch -H Unit1.cpp 
[bcc32 Error] uTPLb_CryptographicLibrary.hpp(174): E2370 Simple type name expected
  Full parser context
    Unit1.cpp(6): #include Unit1.h
    Unit1.h(12): #include c:\program files\embarcadero\studio\16.0\Components\LockBox3\hpp\uTPLb_Codec.hpp
    uTPLb_Codec.hpp(21): #include c:\program files\embarcadero\studio\16.0\Components\LockBox3\hpp\uTPLb_Asymetric.hpp
    uTPLb_Asymetric.hpp(19): #include c:\program files\embarcadero\studio\16.0\Components\LockBox3\hpp\uTPLb_CodecIntf.hpp
    uTPLb_CodecIntf.hpp(21): #include c:\program files\embarcadero\studio\16.0\Components\LockBox3\hpp\uTPLb_CryptographicLibrary.hpp
    uTPLb_CryptographicLibrary.hpp(31): namespace Utplb_cryptographiclibrary
    uTPLb_CryptographicLibrary.hpp(84): class TCryptographicLibrary
Failed
Elapsed time: 00:00:00.5
